game/graphics/opengl_renderer/Sprite3.cpp

```diff
@@ -116,15 +126,94 @@ Sprite3::Sprite3(const std::string& name, BucketId my_id)
+  glBindFramebuffer(GL_FRAMEBUFFER, 0);
+
+  glGenBuffers(1, &m_distort_ogl.vertex_buffer);
+  glGenVertexArrays(1, &m_distort_ogl.vao);
+  glBindVertexArray(m_distort_ogl.vao);
+  glBindBuffer(GL_ARRAY_BUFFER, m_distort_ogl.vertex_buffer);
+  // note: each sprite shares a single vertex per slice, account for that here
+  int distort_vert_buffer_len =
+      SPRITE_RENDERER_MAX_DISTORT_SPRITES *
+      ((5 - 1) * 11 + 1);  // max * ((verts_per_slice - 1) * max_slices + 1)
+  glBufferData(GL_ARRAY_BUFFER, distort_vert_buffer_len * sizeof(SpriteDistortVertex), nullptr,
+               GL_DYNAMIC_DRAW);
+  glEnableVertexAttribArray(0);
+  glVertexAttribPointer(0,                                         // location 0 in the shader
+                        3,                                         // 3 floats per vert
+                        GL_FLOAT,                                  // floats
+                        GL_TRUE,                                   // normalized, ignored,
+                        sizeof(SpriteDistortVertex),               //
+                        (void*)offsetof(SpriteDistortVertex, xyz)  // offset in array
+  );
+  glEnableVertexAttribArray(1);
+  glVertexAttribPointer(1,                                        // location 1 in the shader
+                        3,                                        // 2 floats per vert
+                        GL_FLOAT,                                 // floats
+                        GL_TRUE,                                  // normalized, ignored,
+                        sizeof(SpriteDistortVertex),              //
+                        (void*)offsetof(SpriteDistortVertex, st)  // offset in array
+  );
+
+  // note: add one extra element per sprite that marks the end of a triangle strip
+  int distort_idx_buffer_len = SPRITE_RENDERER_MAX_DISTORT_SPRITES *
+                               ((5 * 11) + 1);  // max * ((verts_per_slice * max_slices) + 1)
+  glGenBuffers(1, &m_distort_ogl.index_buffer);
+  glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, m_distort_ogl.index_buffer);
+  glBufferData(GL_ELEMENT_ARRAY_BUFFER, distort_idx_buffer_len * sizeof(u32), nullptr,
+               GL_DYNAMIC_DRAW);
+
+  glBindBuffer(GL_ARRAY_BUFFER, 0);
+  glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, 0);
+  glBindVertexArray(0);
+
+  m_sprite_distorter_vertices.resize(distort_vert_buffer_len);
+  m_sprite_distorter_indices.resize(distort_idx_buffer_len);
+  m_sprite_distorter_frame_data.resize(SPRITE_RENDERER_MAX_DISTORT_SPRITES);
 }
 
 /*!
- * Run the sprite distorter.  Currently nothing uses sprite-distorter so this just skips through
- * the table upload stuff that runs every frame, even if there are no sprites.
+ * Run the sprite distorter.
  */
 void Sprite3::render_distorter(DmaFollower& dma,
                                SharedRenderState* render_state,
                                ScopedProfilerNode& prof) {
+  // First, make sure the distort framebuffer is the correct size
```

Comment:
It might be nice to add a way to disable sprite distort, just in case it causes massive performance issues for some people.

You could follow the example of `m_3d_enable` to set up a checkbox in the debug GUI.

Reply:
Done, also made it respect the global sprite `m_enabled` flag.